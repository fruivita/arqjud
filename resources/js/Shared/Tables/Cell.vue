<!--
    Célula (td) da tabela padrão.

    @link https://vuejs.org/guide/introduction.html
    @link https://tailwindcss.com/docs
    @link https://inertiajs.com/
 -->

<script setup>
import Tooltip from '@/Shared/Misc/Tooltip.vue';

defineProps({
    erro: { type: String, default: '' },
    tooltip: { type: String },
    tooltip: { type: [Array, String] },
});
</script>

<template>
    <td class="p-3" dusk="slot">
        <div class="flex items-center justify-center">
            <slot></slot>

            <Tooltip v-if="tooltip" :texto="tooltip" class="ml-1" />
        </div>

        <span v-if="erro" class="block text-sm font-bold text-red-500">{{ erro }}</span>
    </td>
</template>
