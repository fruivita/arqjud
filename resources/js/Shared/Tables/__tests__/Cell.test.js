/**
 * Testes para o componente Cell.
 *
 * @link https://vuejs.org/guide/scaling-up/testing.html
 * @link https://test-utils.vuejs.org/guide/
 * @link https://vitest.dev/
 */

import Cell from '@/Shared/Tables/Cell.vue';
import { mount } from '@vue/test-utils';
import { beforeEach, describe, expect, test } from 'vitest';

let mountFunction;

beforeEach(() => {
    mountFunction = (options = {}) => {
        return mount(Cell, { ...options });
    };
});

describe('Cell', () => {
    test('propriedades estÃ£o definidas', () => {
        expect(Cell.props).toMatchObject({
            erro: { type: String, default: '' },
            tooltip: { type: [Array, String] },
        });
    });

    test('renderiza o componente respeitando o snapshot', () => {
        expect(
            mountFunction({
                slots: { default: 'foo' },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza a mensagem de erro respeitando o snapshot', () => {
        expect(
            mountFunction({
                slots: { default: 'foo' },
                props: { erro: 'bar' },
            }).html()
        ).toMatchSnapshot();
    });

    test('renderiza o componente tooltip respeitando o snapshot', () => {
        expect(
            mountFunction({
                slots: { default: 'foo' },
                props: { erro: 'bar', tooltip: 'baz' },
            }).html()
        ).toMatchSnapshot();
    });
});
