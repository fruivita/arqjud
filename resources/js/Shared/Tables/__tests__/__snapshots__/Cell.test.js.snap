// Vitest Snapshot v1

exports[`Cell > renderiza a mensagem de erro respeitando o snapshot 1`] = `
"<td class=\\"p-3\\" dusk=\\"slot\\">
  <div class=\\"flex items-center justify-center\\">foo</div><span class=\\"block text-sm font-bold text-red-500\\">bar</span>
</td>"
`;

exports[`Cell > renderiza de maneira diversa o componente se ele for fixo respeitando o snapshot 1`] = `
"<td class=\\"sticky left-0 z-10 h-fit p-3\\" dusk=\\"slot\\">
  <div class=\\"flex items-center justify-center\\"></div>
  <!--v-if-->
</td>"
`;

exports[`Cell > renderiza o componente respeitando o snapshot 1`] = `
"<td class=\\"p-3\\" dusk=\\"slot\\">
  <div class=\\"flex items-center justify-center\\">foo</div>
  <!--v-if-->
</td>"
`;
